File: front/lib/api/src/models/image.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'image.freezed.dart';
part 'image.g.dart';

@freezed
class Image with _$Image {
  const factory Image({
    required String id,
    required String blurhash,
    @JsonKey(name: "aspect_ratio") required double aspectRatio,
    required List<String> colors,
  }) = _Image;

  factory Image.fromJson(Map<String, dynamic> json) => _$ImageFromJson(json);
}

File: front-vue/models/image.ts
import * as yup from "yup";
import { Resource } from "./resource";

const Image = Resource.concat(
    yup.object({
        blurhash: yup.string().required(),
        aspect_ratio: yup.number().required(),
        colors: yup.array(yup.string().required()).required(),
    }),
);

type Image = yup.InferType<typeof Image>;

export { Image };
